src/scheduler/agent/database.c

```diff
@@ -1132,6 +1118,30 @@ char* get_email_command(scheduler_t* scheduler, char* user_email)
         g_string_append(client_cmd, "warn");
       }
     }
+    if(g_hash_table_contains(smtpvariables, "SMTPAuthUser") && g_hash_table_contains(smtpvariables, "SMTPAuthPasswd")
+          && g_hash_table_contains(smtpvariables, "SMTPHostName") && g_hash_table_contains(smtpvariables, "SMTPPort"))
+    {
+      if(system("s-nail --help") == 0)
+      {
+        g_string_append(client_cmd, " -S v15-compat");
+        g_string_append(client_cmd, " -S mta=\"smtps://");
+        g_string_append(client_cmd, (char *)g_hash_table_lookup(smtpvariables, "SMTPAuthUser"));
+        g_string_append(client_cmd, ":");
+        g_string_append(client_cmd, (char *)g_hash_table_lookup(smtpvariables, "SMTPAuthPasswd"));
```

Comment:
Sorry, I missed it. It will be nice to use `smtps` only if port is not `25` or `SMTPStartTls` is provided. Otherwise, `smtp://` should be good enough as per the s-nail's man page.

Also, will it be possible to do a url encoding on the `username:password`? https://manpages.debian.org/stretch/s-nail/s-nail.1.en.html#On_URL_syntax_and_credential_lookup

Reply:
Did the asked changes.
For URL encoding, I took reference from [here](http://geekhideout.com/urlcode.shtml)

Reply:
@Aman-Codes , it will be fine to have the code from geekhideout. Please provide the link to the website. It is already mentioned to be in Public Domain. (Please do mention the mentioned statement from the webpage: **To the extent possible under law, Fred Bulback has waived all copyright and related or neighboring rights to URL Encoding/Decoding in C. This work is published from: United States.**)

![image](https://user-images.githubusercontent.com/18077542/108711386-4e2ca100-753b-11eb-9cfc-01dd1b3433e4.png)

Reply:
@GMishx Added the link to their [website](http://geekhideout.com/urlcode.shtml) and the mentioned statement from their webpage